spec/e2e/relationship_spec.rb

```diff
@@ -589,4 +589,65 @@ class ActiveRelSpecTypesInheritedRelClass < ActiveRelSpecTypesAutomaticRelType
       end
     end
   end
+
+  context "between nodes of the same class" do
+    before(:each) do
+      clear_model_memory_caches
+
+      stub_node_class('NodeClass') do
+        has_one  :out, :parent,   rel_class: :IsParentOf
+        has_many :in,  :children, rel_class: :IsParentOf
+      end
+
+      stub_relationship_class('IsParentOf') do
+        from_class :NodeClass
+        to_class   :NodeClass
+      end
+
+      @gen0 = NodeClass.create
+      @gen1 = @gen2 = nil
+    end
+
+    after(:each) { [@gen0, @gen1, @gen2].compact.each(&:destroy) }
+
+    shared_examples_for :creating_gen2 do
```

Comment:
This is used just once, Do we need a shared example here?

Reply:
Nope. I was thinking I wanted to use them in more contexts (e.g., `creates_unique true`, etc.), but then didn't.
I'll just roll them into the loop; I/you/we/someone can expand the loop if/when that's desired.

Reply:
26da404